crates/proto/proto/note.proto

```diff
@@ -5,24 +5,33 @@ import "digest.proto";
 import "merkle.proto";
 import "account.proto";
 
+enum NoteType {
+  PUBLIC = 0;
+  OFF_CHAIN = 1;
+  ENCRYPTED = 2;
+}
```

Comment:
Yeah, not sure about this. There cannot be protobuf enums that don't start with 0 so it's either creating a phantom variant, creating a custom mapping function (from protobuf's note type to base note type), or just removing it altogether and using a `fixed32`. For making sure they stay synced I can add a test that asserts this.

Reply:
On second thought it's not worth the added complexity if we can just convert the note type number. Removed in favor of a `fixed32`

Reply:
Agreed! I came to the same conclusion. The only question - should we use `int32` instead? (this way, encoding would take 1 byte instead of 4, but otherwise should be the same).

Reply:
Changed to `uint32`. Seemed similar in terms of encoding low values but marginally more correct for the use case.